blaze/server/serialization.py

```diff
@@ -34,6 +87,101 @@ def _coerce_str(bytes_or_str):
 )
 
 
+try:
+    import blosc
+    del blosc
+    compress = 'blosc'
+except ImportError:
+    compress = None
+
+
+def fastmsgpack_object_hook(ob):
+    typ = ob.get('typ')
+    if typ is None:
+        return ob
+    if typ == 'nat':
+        return pd.NaT
+    return pd_msgpack_object_hook(ob)
+
+
+if PY2:
+    def _l1(bs):
+        if isinstance(bs, unicode):
+            return bs.encode('latin1')
+        return bs
+else:
+    def _l1(bs):
+        return bs
+
+
+def fastmsgpack_data_loads(data):
+    raw = list(msgpack_unpack(
+        BytesIO(_l1(data)),
+        object_hook=fastmsgpack_object_hook,
+    ))
+    if len(raw) == 1:
+        return raw[0]
+    return raw
+
+
+def fastmsgpack_loads(data):
+    raw = list(msgpack_unpack(
+        BytesIO(_l1(data)),
+        object_hook=object_hook,
+    ))
+    if len(raw) == 1:
+        return raw[0]
+    return raw
+
+
+def fastmsgpack_default(ob):
+    if ob is pd.NaT:
+        return {'typ': 'nat'}
+    return pd_msgpack_default(ob)
+
+
+def fastmsgpack_data_dumps(data):
+    return {
+        '__!bytes': pd.to_msgpack(
+            None,
+            data,
+            compress=compress,
+            default=fastmsgpack_default,
+            encoding='latin1',
```

Comment:
when using `utf-8` with a python3 server and a python2 client I got failures in the c++ code for msgpack about a dict not being hashable. I explicitly set this to `latin1` to avoid this. To be fair, the python2 client was using pandas 0.16.1 so it might be fixed on master.